core/src/main/java/org/geysermc/geyser/inventory/holder/BlockInventoryHolder.java

```diff
@@ -83,13 +83,22 @@ public void prepareInventory(InventoryTranslator translator, GeyserSession sessi
                 inventory.setHolderPosition(session.getLastInteractionBlockPosition());
                 ((Container) inventory).setUsingRealBlock(true, javaBlockString[0]);
                 setCustomName(session, session.getLastInteractionBlockPosition(), inventory, javaBlockId);
-                return;
+                return true;
+            }
+        }
+
+        // Check if a fake block can be placed, either above the player or beneath.
+        Vector3i position = session.getPlayerEntity().getPosition().toInt().add(0, 5, 0);
+        if (position.getY() < session.getChunkCache().getMinY()) {
```

Comment:
You should probably use `session.getChunkCache().getBedrockDimension()` instead as the Y values in the world may be different than the server world. (For example: the Nether without the End workaround applied)

Reply:
Ah I already tried it, but misunterstood height as max y, but I needed to use min y + height instead, thanks